### ISSUE TITLE
mda often aborts

### ISSUE THREAD

**Issue by kerschke** [berndbischl/mlr#278]
I just ran into the following problem: ```mda``` often aborts with an error:
**Error in kmeans(xx, start) : initial centers are not distinct**

I tried to make a simple example, when the error occurs:

```{r}
library(mlbench)
data(Satellite)
library(mlr)
set.seed(14 * 10 * 2015)
task = makeClassifTask(data = Satellite, target = "classes")
lrn = makeLearner("classif.mda")
rdesc = makeResampleDesc("Bootstrap", iters = 20L)
rinst = makeResampleInstance(desc = rdesc, task = task)
resample(learner = lrn, task = task, resampling = rinst)
```

I know, that this is not directly an issue of mlr but rather of mda itself, as the following example shows:

```{r}
library(mda)
set.seed(14 * 10 * 2015)
m = replicate(2000, mda(Species ~ ., data = iris))
```

However, as this problem was already mentioned in 2007 (e.g. here: https://stat.ethz.ch/pipermail/r-help/2007-August/138695.html) and apparently the authors of mda so far did not change anything about it, it does not look like they will fix it very soon.

Does anybody have an idea how to bypass that bug?!

**Comment by schiffner** [comment]
I think we should at least try to send a bug report, from the link above it's not clear if the maintainer has been notified directly.

In the meantime we could set the default `start.method` to `"lvq"`(the learner parameter for this is missing at the moment in RLearner_classif_mda.R).
There shouldn't be any problems with identical centers as in `mda:::lvq.start` the `lvqinit` function is used instead of `sample` . I haven't done any further testing, but the above example works with lvq as start method.

```
library(mlbench)
data(Satellite)
library(mlr)
set.seed(14 * 10 * 2015)
task = makeClassifTask(data = Satellite, target = "classes")
configureMlr(on.learner.par.without.desc = "warn")
lrn = makeLearner("classif.mda", start.method = "lvq")
rdesc = makeResampleDesc("Bootstrap", iters = 20L)
rinst = makeResampleInstance(desc = rdesc, task = task)
resample(learner = lrn, task = task, resampling = rinst)
```

**Comment by berndbischl** [comment]
I agree, this is not REALLY an mlr problem. 

1) 
@kerschke Have you sent a short bug report to the authors?

2)
@schiffner 
I have changed the default like you advised. Can you please write whether this affects the method a lot? If not then close.

**Comment by kerschke** [comment]
No, I haven't. I didn't see myself as "we" ;-)

**Comment by berndbischl** [comment]
Well, "we" do not care so much about the error ;-)
I mean, it hurt you so it is in your best interest to report to them

**Comment by kerschke** [comment]
Touché :)

**Comment by schiffner** [comment]
> I have changed the default like you advised. Can you please write whether this affects the method a lot? If not then close.

I don't think that this affects mda much. lvq might possibly have some advantages because it takes the class labels into account.

I can write the bug report tomorrow. (I might have another mda issue to report and then simply can do both in one report.)

**Comment by berndbischl** [comment]
Thx. Closing this here then

**Comment by schiffner** [comment]
Just to let everyone know: The issue concerning initialization by `kmeans` has been solved with mda 0.4-6.
Moreover, I added some missing learner params (14fffe518f181f65c430ad97f70c18a645d0281a).

**Comment by berndbischl** [comment]
Thx!
